query-engine/connectors/sql-connector/src/database/connector_transaction.rs

```diff
@@ -0,0 +1,102 @@
+use prisma_query::connector::Queryable;
+
+use connector_interface::*;
+use prisma_models::*;
+
+use crate::query_builder::write::NestedActions;
+use crate::query_builder::WriteQueryBuilder;
+use crate::transactional;
+use crate::SqlError;
+
+pub struct ConnectorTransaction<'a> {
+    inner: prisma_query::connector::Transaction<'a>,
+}
+
+impl<'a> ConnectorTransaction<'a> {
+    pub fn new(tx: prisma_query::connector::Transaction) -> ConnectorTransaction {
+        ConnectorTransaction { inner: tx }
+    }
+
+    pub fn commit(self) -> crate::Result<()> {
+        Ok(self.inner.commit().map_err(SqlError::from)?)
+    }
+}
+
+impl MaybeTransaction for ConnectorTransaction<'_> {}
+
+impl ReadOperations for ConnectorTransaction<'_> {
+    fn get_single_record(
+        &mut self,
+        record_finder: &RecordFinder,
+        selected_fields: &SelectedFields,
+    ) -> connector_interface::Result<Option<SingleRecord>> {
+        let result = transactional::execute_get_single_record(&mut self.inner, record_finder, selected_fields)?;
+        Ok(result)
+    }
+}
+impl WriteOperations for ConnectorTransaction<'_> {
+    fn create_record(&mut self, model: ModelRef, args: WriteArgs) -> connector_interface::Result<GraphqlId> {
+        let result = transactional::create::execute(&mut self.inner, model, args.non_list_args(), args.list_args())?;
+        Ok(result)
+    }
+
+    fn update_records(
+        &mut self,
+        model: ModelRef,
+        where_: Filter,
+        args: WriteArgs,
+    ) -> connector_interface::Result<usize> {
+        let result = transactional::update_many::execute(
```

Comment:
psst. Please get rid of these modules in transactional and just implement them here. Nicer :)